tests/test_builders.py

```diff
@@ -0,0 +1,146 @@
+from importlib import resources
+from tempfile import NamedTemporaryFile
+
+import numpy as np
+import pytest
+from openmm import app, unit
+
+from timemachine.constants import DEFAULT_PROTEIN_FF, DEFAULT_WATER_FF
+from timemachine.fe.utils import get_romol_conf, set_romol_conf
+from timemachine.ff import sanitize_water_ff
+from timemachine.ff.handlers import openmm_deserializer
+from timemachine.md.barostat.utils import compute_box_volume
+from timemachine.md.builders import build_protein_system, build_water_system
+from timemachine.md.minimizer import check_force_norm
+from timemachine.testsystems.relative import get_hif2a_ligand_pair_single_topology
+
+
+def test_build_water_system():
+    mol_a, mol_b, _ = get_hif2a_ligand_pair_single_topology()
+
+    water_system, water_coords, box, _ = build_water_system(4.0, DEFAULT_WATER_FF)
+
+    water_with_mols, mol_water_coords, box_with_mols, _ = build_water_system(4.0, DEFAULT_WATER_FF, mols=[mol_a, mol_b])
+
+    # No waters should be deleted, but the box will be slightly larger
+    assert len(water_coords) == len(mol_water_coords)
+    assert compute_box_volume(box) < compute_box_volume(box_with_mols)
+
+    box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+    box_with_mols += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+
+    water_system_bps, _ = openmm_deserializer.deserialize_system(water_system, cutoff=1.2)
+    for bp in water_system_bps:
+        (
+            du_dx,
+            _,
+        ) = bp.to_gpu(
+            np.float32
+        ).bound_impl.execute(water_coords, box, compute_u=False)
+        check_force_norm(-du_dx)
+
+    water_system_bps, _ = openmm_deserializer.deserialize_system(water_with_mols, cutoff=1.2)
+    for bp in water_system_bps:
+        (
+            du_dx,
+            _,
+        ) = bp.to_gpu(
+            np.float32
+        ).bound_impl.execute(mol_water_coords, box_with_mols, compute_u=False)
+        check_force_norm(-du_dx)
+
+
+@pytest.mark.nocuda
+def test_build_protein_system_returns_correct_water_count():
+    with resources.path("timemachine.datasets.fep_benchmark.pfkfb3", "6hvi_prepared.pdb") as pdb_path:
+        protein_system, protein_coords, box, _, num_water_atoms = build_protein_system(
+            str(pdb_path), DEFAULT_PROTEIN_FF, DEFAULT_WATER_FF
+        )
+        assert protein_coords.shape[0] - num_water_atoms == 6748
```

Comment:
nit: does 6748 need to be hardcoded here? (If so, could we add a brief comment on how to derive this value from the test data?)

Reply:
It does, as previously this value was wrong. Added a comment explaining why the value is hard coded and how to determine the value: https://github.com/proteneer/timemachine/pull/1346/commits/a86d9b7b516fe99f381a1801a6e901eb4f700f5c